### DISCUSSION TITLE
Querying details of a specific font

### DISCUSSION THREAD

**Discussion by Sadi58** [ImageMagick/ImageMagick#7478] [Help] [answered]
### ImageMagick version

6.9.13.12

### Operating system, version and so on

Debian 13 (Trixie/Testing)

### Description

I get a detailed list of all fonts recognized and supported by ImageMagick with this command:
`identify -list font`
 This gives me various details about these fonts in blocks of text  - which amount to 827 fonts located in both system and user font paths.
However, I would like to retrieve such attributes as "font-weight", "font-style" and "font-stretch" of a font picked via YAD's font selection dialog box, which produces an output like "DejaVu Sans Condensed Bold Oblique 24", so that they can be exported to a CSS style sheet via a shell script.
I just couldn't figure out how to get those details of a specific font instead of getting the whole list, and parsing it every time a font is selected in this way.
The fact that I couldn't even find the command `identify -list font` in the documentation may well mean there's no command like `identify -list "Font-Name"` or some other that can do this, but I just wanted to ask just in case.
Otherwise, maybe this can be considered for future development, as the functionality already exists inside the magic tool.
Thank you.

**Comment by fmw42** [comment]
See https://imagemagick.org/Usage/text/#font_info

**Reply by Sadi58** [reply]
Thank you but this gives font metrics info only.
As I don't have version 7 yet, I can get this info via commands `convert`, `mogrify` and `montage`.
What I'm looking for is what get with the command `identify -list font`.
For example, I expect the output of a command like `magick identify -list font "Arial-Bold-Italic"`:
```
  Font: Arial-Bold-Italic
    family: Arial
    style: Italic
    stretch: Normal
    weight: 700
    glyphs: /usr/share/fonts/truetype/msttcorefonts/arialbi.ttf
```
This is the text block I get for "Arial bold Italic" in a list of 827 fonts generated when I enter the command  `identify -list font`.
If there's nothing like that, I think it would be very useful if the present command can be changed a bit so that while `identify -list font` does that, it can merely display those 4 lines for a specif font when the command is followed by such a font name.

**Reply by fmw42** [reply, marked as answer]
You can pipe to sed to do that

```
identify -list font | sed -n '/Arial-Bold-Italic.*/,+5p'

  Font: Arial-Bold-Italic
    family: Arial
    style: Italic
    stretch: Normal
    weight: 700
    glyphs: /opt/X11/share/system_fonts/Supplemental/Arial Bold Italic.ttf
```

**Reply by Sadi58** [reply]
Thank you very much for this! In deed, I've found this method of piping to sed to be the only way for now, but my sed command is not as good as yours.
Retrieving all font details, and then filtering a specific font is a workaround, of course.
